### PR TITLE
Add custom memory configuration

### PR DESCRIPTION
**Pull request by heidebro**
Add linker scripts both for CMSIS and libopencm3 that increase the size
of the ROM and RAM to 256M for the target board.

Add appropriate Renode configuration files both for the original and
the extended memory configuration.

Amend READMEs accordingly.

Co-authored-by: Lucas Camphausen <lucas.camphausen@iml.fraunhofer.de>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,3 +117,53 @@ This example also comes with a robot test file:
...
+      -DCMAKE_TOOLCHAIN_FILE="`realpath ../build_tools/cmake/arm.toolchain.cmake`" \
+      -DARM_TOOLCHAIN_ROOT="${PATH_TO_ARM_TOOLCHAIN}" \
+      -DARM_CPU="armv7e-m" \
+      -DIREE_HAL_DRIVERS_TO_BUILD="VMVX_Sync" \
+      -DIREE_HOST_BINARY_ROOT="${PATH_TO_IREE_HOST_BINARY_ROOT}" \
+      -DCUSTOM_ARM_LINKER_FLAGS="${CUSTOM_ARM_LINKER_FLAGS}" \
+      -DLINKER_SCRIPT="${PATH_TO_LINKER_SCRIPT}" \
+      ..
+cmake --build . --target sample_vmvx_sync
+```
```
Comment (marbre) [inline README.md:158]: With only `PATH_TO_LINKER_SCRIPT` changed, I don't see a need to replicate the whole instructions. This only makes it harder to keep our docs up-to-date. I would strongly prefer if we can drop the duplication.

File: renode/stm32f4-base.repl [hunk 2]
```diff
@@ -0,0 +1,10 @@
+
+using "platforms/boards/stm32f4_discovery.repl"
+using "platforms/boards/stm32f4_discovery-additional_gpios.repl"
+using "platforms/boards/stm32f4_discovery-bb.repl"
+
+flash: @ sysbus 0x08000000
+    size: 0x100000
+
+sram: @ sysbus 0x20000000
+    size: 0x20000
```
Comment (marbre) [inline renode/stm32f4-base.repl:10]: Missing EOL.

File: renode/stm32f4-base.resc [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+macro reset
+"""
+    sysbus LoadELF $ORIGIN/../build/samples/sample_vmvx_sync
+
+    sysbus WriteDoubleWord 0x1FFF7A10 $id1
+    sysbus WriteDoubleWord 0x1FFF7A14 $id2
+    sysbus WriteDoubleWord 0x1FFF7A18 $id3
+"""
+
+runMacro $reset
```
Comment (marbre) [inline renode/stm32f4-base.resc:32]: Missing EOL.

File: renode/stm32f4-custom.resc [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+macro reset
+"""
+    sysbus LoadELF $ORIGIN/../build/samples/sample_vmvx_sync
+
+    sysbus WriteDoubleWord 0x1FFF7A10 $id1
+    sysbus WriteDoubleWord 0x1FFF7A14 $id2
+    sysbus WriteDoubleWord 0x1FFF7A18 $id3
+"""
+
+runMacro $reset
```
Comment (marbre) [inline renode/stm32f4-custom.resc:32]: Missing EOL.

File: renode/stm32f4-base.resc [hunk 5]
```diff
@@ -0,0 +1,32 @@
+:name: STM32F4 Discovery
+:description: This script runs Contiki on STM32F4 Discovery.
```
Comment (marbre) [inline renode/stm32f4-base.resc:2]: The description seems outdated?

**Review by marbre — CHANGES_REQUESTED** [review]
Some first remarks. I would strongly request to split into several commits as we import several files from different repos. We also need to make sure that license/copyright information is provided correctly. Let's take that discussions f2f.

File: build_tools/stm32f4/stm32f4-discovery-libopencm3-custom.ld [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* Linker script for ST STM32F4DISCOVERY (STM32F407VG, 1024K flash, 128K RAM). */
```
Comment (marbre) [inline build_tools/stm32f4/stm32f4-discovery-libopencm3-custom.ld:21]:
```suggestion
/* Linker script for a custom STM32F4 (STM32F407VG, 256 MB flash, 256 MB RAM). */
```

File: build_tools/stm32f4/stm32f407-cmsis-custom.ld [hunk 7]
```diff
@@ -0,0 +1,365 @@
+/******************************************************************************
+ * @file     stm32f407.ld
```
Comment (marbre) [inline build_tools/stm32f4/stm32f407-cmsis-custom.ld:2]:
```suggestion
 * @file     stm32f407-cmsis-custom.ld
```

File: build_tools/stm32f4/stm32f407-cmsis-custom.ld [hunk 8]
```diff
@@ -0,0 +1,365 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ *  Adapted from the stm32f407.ld linker script of CMSIS.
```
Comment (marbre) [inline build_tools/stm32f4/stm32f407-cmsis-custom.ld:100]:
```suggestion
 *  Adopted from the stm32f407.ld linker script of CMSIS.
```

File: build_tools/stm32f4/stm32f407-cmsis-custom.ld [hunk 9]
```diff
@@ -0,0 +1,365 @@
...
+ *  This script represents common configuration between the linker scripts
+ *  "stm32f407-cmsis-base" and "stm32f407-cmsis-custom".
```
Comment (marbre) [inline build_tools/stm32f4/stm32f407-cmsis-custom.ld:102]: Seems outdated?

**Comment by heidebro** [comment]
Superseded by PR #22

**Pull request closed without merging**